underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystem.java

```diff
@@ -126,6 +126,33 @@ public GCSUnderFileSystem(AlluxioURI uri) throws ServiceException {
     mBucketMode = GCSUtils.translateBucketAcl(acl, mAccountOwnerId);
   }
 
+  /**
+   * Constructor used in test case only.
```

Comment:
Could we unify the code paths for tests and production so that they can be maintained together? We could do this by having the other constructor call this one using `this(...)`. Then we could think of this constructor as being the "real" constructor which follows dependency injection, and the other constructor as being a convenience constructor which creates reasonable default parameters given just an `AlluxioURI`.